api/src/shared/infrastructure/repositories/user-repository.js

```diff
@@ -225,37 +223,24 @@ const updateWithEmailConfirmed = function ({
   return query;
 };
 
-const checkIfEmailIsAvailable = function (email) {
-  return BookshelfUser.query((qb) => qb.whereRaw('LOWER("email") = ?', email.toLowerCase()))
-    .fetch({ require: false })
-    .then((user) => {
-      if (user) {
-        return Promise.reject(new AlreadyRegisteredEmailError());
-      }
+const checkIfEmailIsAvailable = async function (email) {
+  const existingUserEmail = await knex('users').whereRaw('LOWER("email") = ?', email.toLowerCase()).first();
 
-      return Promise.resolve(email);
-    });
+  if (existingUserEmail) throw new AlreadyRegisteredEmailError();
+
+  return email;
 };
 
-const isUserExistingByEmail = function (email) {
-  return BookshelfUser.where({ email: email.toLowerCase() })
-    .fetch()
-    .then(() => true)
-    .catch(() => {
-      throw new UserNotFoundError();
-    });
+const isUserExistingByEmail = async function (email) {
+  const existingUser = await knex('users').where('email', email.toLowerCase()).first();
+  if (!existingUser) throw new UserNotFoundError();
+  return true;
 };
 
-const updateEmail = function ({ id, email }) {
-  return BookshelfUser.where({ id })
-    .save({ email }, { patch: true, method: 'update' })
-    .then((bookshelfUser) => _toDomain(bookshelfUser))
-    .catch((err) => {
-      if (err instanceof BookshelfUser.NoRowsUpdatedError) {
-        throw new UserNotFoundError(`User not found for ID ${id}`);
-      }
-      throw err;
-    });
+const updateEmail = async function ({ id, email }) {
+  const [updatedUserEmail] = await knex('users').where({ id }).update({ email }).returning('*');
```

Comment:
Bookshelf pouvait se charger de modifier automatiquement le `updatedAt` d'un enregistrement, mais avec Knex c'est à faire manuellement à chaque requête effectuant une modification :

```suggestion
  const [updatedUserEmail] = await knex('users').where({ id }).update({ email, updatedAt: new Date() }).returning('*');
```